internal/utils/comparators/versions.go

```diff
@@ -3,27 +3,39 @@ package comparators
 import (
 	"strconv"
 	"strings"
-
-	"github.com/ViaQ/logerr/log"
 )
 
-// CompareVersions will return one of:
+type Version string
+
+func (v Version) ToArray() ([]int, error) {
+	sections := strings.Split(string(v), ".")
+	versionNumbers := make([]int, len(sections))
+
+	for i, s := range sections {
+		vn, err := strconv.Atoi(s)
+		if err != nil {
+			return nil, err
+		}
+		versionNumbers[i] = vn
+	}
+
+	return versionNumbers, nil
+}
+
+// CompareVersionArrays will return one of:
 // -1 : if lhs > rhs
 // 0  : if lhs == rhs
 // 1  : if rhs > lhs
-func CompareVersions(lhs, rhs string) int {
-	lVersions := buildVersionArray(lhs)
-	rVersions := buildVersionArray(rhs)
-
-	lLen := len(lVersions)
-	rLen := len(rVersions)
+func CompareVersionArrays(lhs, rhs []int) int {
```

Comment:
Why did you pick this implementation which splits the original compare up into two functions (parse and compare) instead of just having this function return an error?

Reply:
I picked it this way to make it more clear which version string is the one causing the problems.